resources/manifests/psp-restricted.yaml

```diff
@@ -30,10 +30,7 @@ spec:
   hostPID: false
   runAsUser:
     # Require the container to run without root privileges.
-    rule: 'MustRunAs'
-    ranges:
-    - min: 1
-      max: 99999
+    rule: 'MustRunAsNonRoot'
```

Comment:
From the doc:

> MustRunAsNonRoot - Requires that the pod be submitted with a non-zero runAsUser or have the USER directive defined (using a numeric UID) in the image. Pods which have specified neither runAsNonRoot nor runAsUser settings will be mutated to set runAsNonRoot=true, thus requiring a defined non-zero numeric USER directive in the container. No default provided.

Just ignorant here, but I guess the previous `MustRunAs` also does a mutation if no USER is specified in the pod? I gues it does, as if the pod does not containt that, there is no way other than mutating the spec to add the user information. But just to double check :-)

Asking because PSP policy order is sensible to mutations and will pick one which doesn't need to mutate it if possible. So, it would be nice if the restricted can be used without mutations whenever possible.

Reply:
so earlier each container would start with UIDs allowed between `1` and `99999` if user specified that in either Dockerfile or in pod spec.

If user did not provide UID in either of the two places then the pod spec is mutated to have UID `1`.

---

Now with this change we allow anything as user be it numeric or non-numeric, also we are not limited to upper limit of `99999`, now.

Reply:
> PS: Sorry, I was not assigned as a reviewer and haven't seen it before your ping :)

No problem Rodrigo :+1:

Reply:
Great, so in both cases (before and after this patch) it is mutated if user is not specified in Dockerfile nor pod spec. Seems great, then :)